assets/js/force-2fa.js

```diff
@@ -0,0 +1,88 @@
+/* global ajaxurl, jQuery */
+
+/**
+ * Checks that an element has a non-empty `name` and `value` property.
+ *
+ * @param  {Element} element The element to check
+ * @return {Boolean} true if the element is an input, false if not
+ */
+var isValidElement = function( element ) {
+	return element.name && element.value;
+};
+
+/**
+ * Checks if an element’s value can be saved (e.g. not an unselected checkbox).
+ *
+ * @param  {Element} element The element to check
+ * @return {Boolean} true if the value should be added, false if not
+ */
+var isValidValue = function( element ) {
+	return ( ! [ 'checkbox', 'radio' ].includes( element.type ) || element.checked );
+};
+
+/**
+ * Checks if an input is a checkbox, because checkboxes allow multiple values.
+ *
+ * @param  {Element} element The element to check
+ * @return {Boolean} true if the element is a checkbox, false if not
+ */
+var isCheckbox = function( element ) {
+	return 'checkbox' === element.type;
+};
+
+/**
+ * Retrieves input data from a form and returns it as a JSON object.
+ *
+ * @param  {HTMLFormControlsCollection} elements the form elements
+ * @return {Object} form data as an object literal
+ */
+var formToJSON = function( elements ) {
+	return [].reduce.call( elements, function( data, element ) {
```

Comment:
jQuery has [`serializeArray()`](https://api.jquery.com/serializeArray/) -- could we re-use that instead?